### PR TITLE
Fixes #26947 - correct metadata gen logic

### PR DESCRIPTION
**Pull request by jlsherrill**
a logic check was introduced, but during the
planning phase, the value is actually unresolved
output from a previous step which is always truthy
during the plan phase

### CODE REVIEW

**Comment by jlsherrill** [comment]
To test this, 
1.  create a yum repo 'foo'
2.  add it to a NEW content view named 'bar', publish
3.  look on the filesystem for:  /var/lib/pulp/published/yum/https/repos/Default_Organization/Library/bar
4.  On master, it will not exist.  To repeat this delete the content view version completely (so there are no versions), and repeat step 2)

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

File: test/actions/katello/repository/metadata_generate_test.rb [hunk 1]
```diff
@@ -72,5 +72,17 @@ module Actions
       assert_action_planed_with(action, pulp_publish_class, yum_repo, SmartProxy.pulp_master,
                                 yum_action_options)
     end
+
+    it 'plans a yum refresh with matching content true, but during repository creation mode' do
```
Comment (jturel) [inline test/actions/katello/repository/metadata_generate_test.rb]: This is a valid test case but I don't think it's the one described in the fix (it passes when I revert your code change). Should there be a test to assert that a publish action wasn't planned with the `not_falsey `value or something similar?

Comment (jlsherrill) [reply test/actions/katello/repository/metadata_generate_test.rb]: Good eye :+1:

Comment (jlsherrill) [reply test/actions/katello/repository/metadata_generate_test.rb]: updated and added a 2nd test

**Comment by jturel** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

**Pull request merged**